nft/src/benchmarking.rs

```diff
@@ -193,6 +193,76 @@ benchmarks! {
 		assert_eq!(NFT::<T>::nfts(benchmark_data.nft_id).unwrap().collection_id, Some(benchmark_data.collection_id));
 		assert_eq!(NFT::<T>::collections(benchmark_data.collection_id).unwrap().nfts.contains(&benchmark_data.nft_id), true);
 	}
+
+	create_secret_nft {
+		let s in 0 .. T::CollectionSizeLimit::get() - 1;
+		let benchmark_data = prepare_benchmarks::<T>();
+		let alice: T::AccountId = get_account::<T>("ALICE");
+		let nft_offchain_data: BoundedVec<u8, T::NFTOffchainDataLimit> = BoundedVec::try_from(vec![1; T::NFTOffchainDataLimit::get() as usize]).unwrap();
+		let secret_offchain_data: BoundedVec<u8, T::NFTOffchainDataLimit> = BoundedVec::try_from(vec![1; T::NFTOffchainDataLimit::get() as usize]).unwrap();
+		// Fill the collection.
+		NFT::<T>::create_filled_collection(alice.clone(), benchmark_data.collection_id, 0, s).unwrap();
+	}: _(origin::<T>("ALICE"), nft_offchain_data, secret_offchain_data, PERCENT_100, Some(benchmark_data.collection_id), false)
+	verify {
+		// Get The NFT id.
+		let nft_id = NFT::<T>::next_nft_id() - 1;
+		// Get The NFT.
+		let nft = NFT::<T>::nfts(nft_id).unwrap();
+		// Get the secret offchain_data
+		let secret_offchain_data = NFT::<T>::secret_nfts_offchain_data(nft_id);
+		assert_eq!(nft.owner, alice);
+		assert_eq!(NFT::<T>::collections(benchmark_data.collection_id).unwrap().nfts.contains(&nft_id), true);
+		assert_eq!(nft.collection_id, Some(benchmark_data.collection_id));
+		assert_eq!(nft.state.is_secret, true);
+		assert_eq!(nft.state.is_secret_synced, false);
+		assert!(secret_offchain_data.is_some());
+	}
+
+	convert_to_secret {
+		let benchmark_data = prepare_benchmarks::<T>();
```

Comment:
Check the difference in weight by executing this benchmark with 0 and 100k NFTs in the db.